tools/tools/nanobsd/dhcpd/common

```diff
@@ -137,102 +147,103 @@ WITHOUT_INSTALLLIB=true
-	umount ${NANO_WORLDDIR}/usr/ports/packages
-	rmdir ${NANO_WORLDDIR}/usr/ports/packages
-	rmdir ${NANO_WORLDDIR}/usr/ports/distfiles
-	rmdir ${NANO_WORLDDIR}/usr/ports
+	mkdir -p "${NANO_OBJ}"/ports/distfiles
+	mkdir -p "${NANO_OBJ}"/ports/packages
+	mkdir -p "${NANO_WORLDDIR}"/usr/ports/packages
+	mkdir -p "${NANO_WORLDDIR}"/usr/ports/distfiles
+	mount -t nullfs -o noatime "${NANO_OBJ}"/ports/packages \
+	    "${NANO_WORLDDIR}"/usr/ports/packages
+	mount -t nullfs -o noatime "${NANO_OBJ}"/ports/distfiles \
+	    "${NANO_WORLDDIR}"/usr/ports/distfiles
+	CR env ASSUME_ALWAYS_YES=YES SIGNATURE_TYPE=none /usr/sbin/pkg add /usr/ports/packages/All/"$1".txz
+	umount "${NANO_WORLDDIR}"/usr/ports/distfiles
+	umount "${NANO_WORLDDIR}"/usr/ports/packages
+	rmdir "${NANO_WORLDDIR}"/usr/ports/packages
+	rmdir "${NANO_WORLDDIR}"/usr/ports/distfiles
+	rmdir "${NANO_WORLDDIR}"/usr/ports
 	set +x
 }
 
 # Build a port (with the side effect of creating a package)
-do_add_port ()
+do_add_port()
 {
 	local port_path
 	port_path=$1
 	shift
 	set -x
 	# Need to create ${NANO_OBJ}/ports in this add_port_${port} function
-	mkdir -p ${NANO_OBJ}/ports/distfiles
-	mkdir -p ${NANO_OBJ}/ports/packages
-	mkdir -p ${NANO_PORTS}/packages
-	mkdir -p ${NANO_PORTS}/distfiles
-	mkdir -p ${NANO_WORLDDIR}/usr/src
-	mkdir -p ${NANO_WORLDDIR}/usr/ports
-      	mount -t nullfs -o noatime ${NANO_SRC} ${NANO_WORLDDIR}/usr/src
-	mount -t nullfs -o noatime ${NANO_PORTS} ${NANO_WORLDDIR}/usr/ports
-	mount -t nullfs -o noatime ${NANO_OBJ}/ports/packages \
-	    ${NANO_WORLDDIR}/usr/ports/packages
-	mount -t nullfs -o noatime ${NANO_OBJ}/ports/distfiles \
-	    ${NANO_WORLDDIR}/usr/ports/distfiles
-	mkdir -p ${NANO_WORLDDIR}/dev
-	mount -t devfs devfs ${NANO_WORLDDIR}/dev
-	mkdir -p ${NANO_WORLDDIR}/usr/workdir
-	cp /etc/resolv.conf ${NANO_WORLDDIR}/etc/resolv.conf
+	mkdir -p "${NANO_OBJ}"/ports/distfiles
+	mkdir -p "${NANO_OBJ}"/ports/packages
+	mkdir -p "${NANO_PORTS}"/packages
+	mkdir -p "${NANO_PORTS}"/distfiles
+	mkdir -p "${NANO_WORLDDIR}"/usr/src
+	mkdir -p "${NANO_WORLDDIR}"/usr/ports
+	mount -t nullfs -o noatime "${NANO_SRC}" "${NANO_WORLDDIR}"/usr/src
+	mount -t nullfs -o noatime "${NANO_PORTS}" "${NANO_WORLDDIR}"/usr/ports
+	mount -t nullfs -o noatime "${NANO_OBJ}"/ports/packages \
+	    "${NANO_WORLDDIR}"/usr/ports/packages
+	mount -t nullfs -o noatime "${NANO_OBJ}"/ports/distfiles \
+	    "${NANO_WORLDDIR}"/usr/ports/distfiles
+	mkdir -p "${NANO_WORLDDIR}"/dev
+	mount -t devfs devfs "${NANO_WORLDDIR}"/dev
+	mkdir -p "${NANO_WORLDDIR}"/usr/workdir
+	cp /etc/resolv.conf "${NANO_WORLDDIR}"/etc/resolv.conf
 	# OK, a little inefficient, but likely not enough to worry about.
 	CR ldconfig /lib /usr/lib /usr/local/lib
 	CR ldconfig -R
 	CR ldconfig -r
-# Improvement: Don't know why package-recursive don't works here
-	CR "env UNAME_p=${NANO_ARCH} TARGET=${NANO_ARCH} \
+# Improvement: Don't know why package-recursive don't work here
+	CR "env UNAME_p=${NANO_ARCH} TARGET=${NANO_TARGET} \
```

Comment:
`UNAME_p` is not used. Try removing `TARGET`, `TARGET_ARCH`, we are not cross-building.